src/scenes/Event/store.js

```diff
@@ -476,6 +553,131 @@ export default class {
+  getBetAndVoteBalances = async () => {
+    const { address: contractAddress, app: { wallet } } = this;
+
+    // Get all votes for this Topic
+    const voteFilters = [];
+    _.each(wallet.addresses, (item) => {
+      voteFilters.push({
+        topicAddress: contractAddress,
+        voterQAddress: item.address,
+      });
+    });
+
+    // Filter unique votes
+    const allVotes = await queryAllVotes(voteFilters);
+    const uniqueVotes = [];
+    _.each(allVotes, (vote) => {
+      const { voterQAddress, topicAddress } = vote;
+      if (!_.find(uniqueVotes, { voterQAddress, topicAddress })) {
+        uniqueVotes.push(vote);
+      }
+    });
+
+    // Call bet and vote balances for each unique vote address and get arrays for each address
+    const betArrays = [];
+    const voteArrays = [];
+    for (let i = 0; i < uniqueVotes.length; i++) {
+      const voteObj = uniqueVotes[i];
+
+      const betBalances = await axios.post(networkRoutes.api.betBalances, { // eslint-disable-line
+        contractAddress,
+        senderAddress: voteObj.voterQAddress,
+      });
+      betArrays.push(_.map(betBalances.data.result[0], satoshiToDecimal));
+
+      const voteBalances = await axios.post(networkRoutes.api.voteBalances, { // eslint-disable-line
+        contractAddress,
+        senderAddress: voteObj.voterQAddress,
+      });
+      voteArrays.push(_.map(voteBalances.data.result[0], satoshiToDecimal));
+    }
+
+    // Sum all arrays by index into one array
+    const bets = _.map(_.unzip(betArrays), _.sum);
+    const votes = _.map(_.unzip(voteArrays), _.sum);
+    return { bets, votes };
+  }
+
+  getEscrowAmount = async () => {
+    const escrowRes = await axios.post(networkRoutes.api.eventEscrowAmount, {
+      senderAddress: this.app.wallet.lastUsedAddress,
+    });
+    const eventEscrowAmount = satoshiToDecimal(escrowRes.data.result[0]);
+    return eventEscrowAmount;
+  }
+
+  getWithdrawableAddresses = async () => {
+    const { address: eventAddress, app: { wallet } } = this;
+    const withdrawableAddresses = [];
+
+    // Fetch Topic
+    const topics = await queryAllTopics([{ address: eventAddress }]);
+    let topic;
+    if (!_.isEmpty(topics)) {
+      topic = processTopic(topics[0]);
+    } else {
+      throw new Error(`Unable to find topic ${eventAddress}`);
+    }
+
+    // Get all winning votes for this Topic
+    const voteFilters = [];
+    let escrowClaim = 0; // eslint-disable-line
+    _.each(wallet.addresses, (item) => {
+      voteFilters.push({
+        topicAddress: topic.address,
+        optionIdx: topic.resultIdx,
+        voterQAddress: item.address,
+      });
+
+      // Add escrow withdraw object if is event creator
+      if (item.address === topic.creatorAddress) {
+        withdrawableAddresses.push({
+          type: TransactionType.WITHDRAW_ESCROW,
+          address: item.address,
+          botWon: topic.escrowAmount,
+          qtumWon: 0,
+        });
+        escrowClaim = topic.escrowAmount;
```

Comment:
need to set this as a store var and use it.

Reply:
not really sure what that means, I need to spend time on understanding `escrowAmount` and `escrowClaim`, looks like `escrowClaim` is in every address in the withdrawableAddresses. hm

Reply:
updated